jobs/build.Jenkinsfile

```diff
@@ -230,9 +230,38 @@ lock(resource: "build-${params.STREAM}") {
 
         // fetch from repos for the current build
         stage('Fetch') {
+            def overrides_fetch_param = ""
+            # Dont run this for production builds
+            recent_commits_to_lockfiles() {
+                // Checking if last commits meet the criteria
+                shwrap("""
+                # Last author time in UTC
+                echo "Last commit: $(date -ud @$(git log -1 --follow manifest-lock* --format="%ct"))" 
+                # Number of minutes to check
+                n=120
+                # Current time
+                current_time=$(date +%s)
+                # Time n minutes ago
+                n_minutes_ago=$(( current_time - ( 60 * n ) ))
+                # Last commit
+                last_git_log=$(git log -1 --follow manifest-lock* --format="%cI")
+                # Check if git log was run within the last n minutes
+                if (( last_git_log > n_minutes_ago )); then
+                    echo "Last commit was authored less than $n minutes ago."
+                    # Workaround implemented, use --with-cosa-overrides
+                    return 0
```

Comment:
@dustymabe I tried to implement your suggestion, but not sure if this is going to work, so pushing it just for further discussion.